July 2025 release notes (#2262)
semgrep/semgrep-docs: release-notes/july-2025.md

- Fixed an issue where error details weren't printed when an `SemgrepError` exception caused `semgrep` to fail.
- Semgrep now prints an error message and exits instead of silently exiting with code `2` when you run `semgrep scan` in a Docker container without an argument, and there's no target project mounted under `/src`.
- Fixed an issue where a `Unix.Unix_error` would occasionally crash the experimental language server on startup.
- Fixed an issue where scans of large repositories in debug mode resulted in overly large logs.
- Path filters, such as `paths.exclude` and `paths.include` in rules, now apply to normalized file paths relative to the project rule. This makes rule selection independent of the current work folder.
- Patterns with a leading slash, such as `/src`, are now anchored instead
of floating. For example, `exclude: [ "/src" ]` excludes the target
file `src/main.c`, but not `misc/src/main.c`
- **Java**: deprecated the `class $A` partial class pattern in favor of `class $A { ... }`.
- **Python**: Fixed an issue where the Python parser didn't correctly parse and handle valid structural dictionary patterns.

## ⛓️ Semgrep Supply Chain

### Added

> - Supply Chain support for PHP is now **generally available (GA)**.

Reviewer: see comment above about reachability

> - You can now use the **Upgrade guidance** filter to look for findings based on Assistant's analysis and recommendations of whether upgrading to the dependency that is recommended to remediate the vulnerability.

Reviewer: ```suggestion
- You can now use the **Upgrade guidance** filter to look for findings based on whether upgrading to the dependency that remediates the vulnerability introduces breaking changes or not.
```

Reviewer: I found the sentence a bit hard to read so I added a suggestion. Feel free to continue editing it though.
